src/Microsoft.Health.Dicom.Functions/Indexing/ReindexOperation.cs

```diff
@@ -0,0 +1,141 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.Core.Features.Indexing;
+using Microsoft.Health.Dicom.Functions.Indexing.Configuration;
+using Microsoft.Health.Dicom.Functions.Indexing.Models;
+
+namespace Microsoft.Health.Dicom.Functions.Indexing
+{
+    /// <summary>
+    /// Represents the Azure Durable Functions that perform the re-indexing of previously added DICOM instances
+    /// based on new tags configured by the user.
+    /// </summary>
+    public partial class ReindexOperation
+    {
+        private readonly ReindexConfiguration _reindexConfig;
+        private readonly ITagReindexOperationStore _tagOperationStore;
+        private readonly IInstanceReindexer _instanceReindexer;
+        private readonly IAddExtendedQueryTagService _addExtendedQueryTagService;
+
+        public ReindexOperation(IOptions<IndexingConfiguration> configOptions,
+            IAddExtendedQueryTagService addExtendedQueryTagService,
+            ITagReindexOperationStore tagOperationStore,
+            IInstanceReindexer instanceReindexer)
+        {
+            EnsureArg.IsNotNull(configOptions, nameof(configOptions));
+            EnsureArg.IsNotNull(tagOperationStore, nameof(tagOperationStore));
+            EnsureArg.IsNotNull(instanceReindexer, nameof(instanceReindexer));
+            EnsureArg.IsNotNull(addExtendedQueryTagService, nameof(addExtendedQueryTagService));
+            _reindexConfig = configOptions.Value.Add;
+            _tagOperationStore = tagOperationStore;
+            _instanceReindexer = instanceReindexer;
+            _addExtendedQueryTagService = addExtendedQueryTagService;
+        }
+
+        /// <summary>
+        ///  The orchestration function to add extended query tags.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="logger">The logger.</param>
+        /// <returns>The task.</returns>
+        [FunctionName(nameof(AddExtendedQueryTagsOrchestrationAsync))]
```

Comment:
I think we can shorten these names a bit, like `AddExtendedQueryTagAsync` for the function names

Reply:
AddExtendedQueryTagsAsync is occupied by the httpTrigger...
The naming always a bit headache for me, Orchestration is really long word -- maybe AddExtendedQueryTagsOrcAsync?

Reply:
Well, we could rename the HTTP trigger too. It's technically not "adding," but rather "starting." So perhaps the HTTP trigger is `StartExtendedQueryTagAdditionAsync` while the orchestration is `AddExtendedQueryTagAsync`?

Reply:
or `StartAddingExtendedQueryTagAsync`?